[2025-05-21T00:21:55.808+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-05-21T00:21:55.856+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: merge_sources_pipeline.prediction manual__2025-05-21T00:11:28.675542+00:00 [queued]>
[2025-05-21T00:21:55.876+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: merge_sources_pipeline.prediction manual__2025-05-21T00:11:28.675542+00:00 [queued]>
[2025-05-21T00:21:55.876+0000] {taskinstance.py:2867} INFO - Starting attempt 2 of 2
[2025-05-21T00:21:55.900+0000] {taskinstance.py:2890} INFO - Executing <Task(_PythonDecoratedOperator): prediction> on 2025-05-21 00:11:28.675542+00:00
[2025-05-21T00:21:55.921+0000] {standard_task_runner.py:72} INFO - Started process 592 to run task
[2025-05-21T00:21:55.931+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'merge_sources_pipeline', 'prediction', 'manual__2025-05-21T00:11:28.675542+00:00', '--job-id', '74', '--raw', '--subdir', 'DAGS_FOLDER/inference_dag.py', '--cfg-path', '/tmp/tmpqyr2onjj']
[2025-05-21T00:21:55.938+0000] {standard_task_runner.py:105} INFO - Job 74: Subtask prediction
[2025-05-21T00:21:56.036+0000] {task_command.py:467} INFO - Running <TaskInstance: merge_sources_pipeline.prediction manual__2025-05-21T00:11:28.675542+00:00 [running]> on host 7f42758d044f
[2025-05-21T00:21:56.226+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='merge_sources_pipeline' AIRFLOW_CTX_TASK_ID='prediction' AIRFLOW_CTX_EXECUTION_DATE='2025-05-21T00:11:28.675542+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-21T00:11:28.675542+00:00'
[2025-05-21T00:21:56.229+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-05-21T00:21:56.273+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 266, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/dags/inference_dag.py", line 68, in prediction
    result_df = run_inference(preprocessed_df)
  File "/opt/airflow/scripts/inference.py", line 16, in run_inference
    y_pred = predictor.predict_contrails(df_predict)
  File "/opt/airflow/scripts/predictor.py", line 33, in predict_contrails
    return self.model.predict(df)
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/core.py", line 729, in inner_f
    return func(**kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/sklearn.py", line 1327, in predict
    predts = self.get_booster().inplace_predict(
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/core.py", line 729, in inner_f
    return func(**kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/core.py", line 2667, in inplace_predict
    self._validate_features(fns)
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/core.py", line 3243, in _validate_features
    raise ValueError(msg.format(self.feature_names, feature_names))
ValueError: feature_names mismatch: ['Aircraft', 'Engine', 'Seats', 'Origin Airport', 'Destination Airport', 'Distance Flown (km)', 'CO2 (kgCO2e)', 'Month', 'Day', 'Weekday', 'Takeoff_Hour'] []
expected Engine, Month, Distance Flown (km), Weekday, Origin Airport, CO2 (kgCO2e), Destination Airport, Aircraft, Day, Seats, Takeoff_Hour in input data
[2025-05-21T00:21:56.297+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=merge_sources_pipeline, task_id=prediction, run_id=manual__2025-05-21T00:11:28.675542+00:00, execution_date=20250521T001128, start_date=20250521T002155, end_date=20250521T002156
[2025-05-21T00:21:56.353+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-05-21T00:21:56.354+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 74 for task prediction (feature_names mismatch: ['Aircraft', 'Engine', 'Seats', 'Origin Airport', 'Destination Airport', 'Distance Flown (km)', 'CO2 (kgCO2e)', 'Month', 'Day', 'Weekday', 'Takeoff_Hour'] []
expected Engine, Month, Distance Flown (km), Weekday, Origin Airport, CO2 (kgCO2e), Destination Airport, Aircraft, Day, Seats, Takeoff_Hour in input data; 592)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 266, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/dags/inference_dag.py", line 68, in prediction
    result_df = run_inference(preprocessed_df)
  File "/opt/airflow/scripts/inference.py", line 16, in run_inference
    y_pred = predictor.predict_contrails(df_predict)
  File "/opt/airflow/scripts/predictor.py", line 33, in predict_contrails
    return self.model.predict(df)
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/core.py", line 729, in inner_f
    return func(**kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/sklearn.py", line 1327, in predict
    predts = self.get_booster().inplace_predict(
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/core.py", line 729, in inner_f
    return func(**kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/core.py", line 2667, in inplace_predict
    self._validate_features(fns)
  File "/home/airflow/.local/lib/python3.10/site-packages/xgboost/core.py", line 3243, in _validate_features
    raise ValueError(msg.format(self.feature_names, feature_names))
ValueError: feature_names mismatch: ['Aircraft', 'Engine', 'Seats', 'Origin Airport', 'Destination Airport', 'Distance Flown (km)', 'CO2 (kgCO2e)', 'Month', 'Day', 'Weekday', 'Takeoff_Hour'] []
expected Engine, Month, Distance Flown (km), Weekday, Origin Airport, CO2 (kgCO2e), Destination Airport, Aircraft, Day, Seats, Takeoff_Hour in input data
[2025-05-21T00:21:56.388+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-05-21T00:21:56.412+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
